src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -631,7 +747,12 @@ ss::future<> ntp_archiver::sync_manifest_until_abort() {
                 vlog(
                   _rtclog.debug,
                   "sync manifest loop waiting for leadership/unpause");
-                co_await _leader_cond.wait();
+                constexpr auto leader_cond_timeout = 30s;
+                _rtctx.suspend_to(
+                  ss::lowres_clock::now() + leader_cond_timeout);
```

Comment:
With very low probability but we have a potential race here where `now() + leader_cond_timeout` will match almost exactly with `_stall_timeout` periodic timer and we will have a chance of triggering the warning?

We should account for that to avoid noise.